### PR TITLE
TEIIDSB-170: rolling back the parsing of the database, and allow mult…

### PR DESCRIPTION
**Pull request by rareddy**
…i-comments where white space can be defined, added test cases for same

### CODE REVIEW

**Comment by rareddy** [comment]
@shawkins removed the more contextual parsing now, allowed support of comments

**Comment by shawkins** [comment]
Yes that looks like it will catch multi-line comments without nested comments.  End of line comments would still be an issue.

**Comment by rareddy** [comment]
If you can suggest the updated regex that will be good, otherwise that is best I could come up with.

**Comment by shawkins** [comment]
> If you can suggest the updated regex that will be good, otherwise that is best I could come up with.

Without multi-line nesting (which is generally only used for source hints), you have something like:

`(/\\*([^*]|\\*[^/])*\\*/|--[^\r\n]*[\r\n]|\\s)+`

File: pkg/util/vdbutil/parser.go [hunk 1]
```diff
@@ -41,74 +43,49 @@ func ParseDataSourcesInfoFromDdl(ddl string) []DatasourceInfo {
...
-	databaseEx := "CREATE\\s+DATABASE\\s+" + id + ";?\\s+.*"
-	databaseRegEx := regexp.MustCompile(databaseEx)
-
-	createSchemaEx := "CREATE\\s+VIRTUAL\\s+SCHEMA\\s+" + id + "[;?$|\\s+.*]"
-	createSchemaRegEx := regexp.MustCompile(createSchemaEx)
 
-	setSchemaEx := "SET\\s+SCHEMA\\s+" + id + "[;?$|\\s+.*]"
-	setSchemaRegEx := regexp.MustCompile(setSchemaEx)
+	commentOrSpace := "(/\\*([^*]|[\r\n]|(\\*([^/]|[\r\n])))*\\*/|\\s)+"
+	viewEx := "CREATE" + commentOrSpace + "VIEW" + commentOrSpace
```
[the remark this answers is not in the document]
Comment (rareddy) [reply pkg/util/vdbutil/parser.go:73]: I will see if I can update the regex to allow above nested case.

File: pkg/util/vdbutil/parser_test.go [hunk 2]
```diff
@@ -98,16 +123,52 @@ func TestMaterialized(t *testing.T) {
...
 		"teiid_rel:ALLOW_MATVIEW_MANAGEMENT" 'true'
 	) AS 
 		select t.*, AVG("close") OVER (ORDER BY "date" ASC ROWS 9 PRECEDING) AS MA10 from foo;		  
 	`
-	assert.Equal(t, 2, len(MaterializiedViewsInDdl(ddl)))
-	assert.Equal(t, []string{"customer.portfolio.vix", "customer.portfolio.vix2"}, MaterializiedViewsInDdl(ddl))
+	assert.Equal(t, true, ShouldMaterialize(ddl))
+}
+
+func TestComment(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (rareddy) [reply pkg/util/vdbutil/parser_test.go]: removed the test

File: pkg/util/vdbutil/parser.go [hunk 3]
```diff
@@ -41,74 +43,49 @@ func ParseDataSourcesInfoFromDdl(ddl string) []DatasourceInfo {
...
 		}
 	}
 	return sources
 }
 
-// MaterializiedViewsInDdl --
-func MaterializiedViewsInDdl(ddl string) []string {
+// ShouldMaterialize --
+func ShouldMaterialize(ddl string) bool {
 	materialization := "MATERIALIZED\\s+'?TRUE'?"
```
[the remark this answers is not in the document]
Comment (rareddy) [reply pkg/util/vdbutil/parser.go]: updated now

File: pkg/util/vdbutil/parser.go [hunk 4]
```diff
@@ -41,74 +43,49 @@ func ParseDataSourcesInfoFromDdl(ddl string) []DatasourceInfo {
 
 	var sources []DatasourceInfo
 	id := "(\\w+|(?:\"[^\"]*\")|()'[^']*'+)"
-	regEx := "CREATE\\s+SERVER\\s+" + id + "\\s+(TYPE\\s+" + id + "\\s+)??FOREIGN\\s+DATA\\s+WRAPPER\\s+" + id
+	commentOrSpace := "(/\\*([^*]|[\r\n]|(\\*([^/]|[\r\n])))*\\*/|\\s)+"
+	serverRegEx := "CREATE" + commentOrSpace + "SERVER" + commentOrSpace + id + commentOrSpace + "(TYPE" + commentOrSpace + id + commentOrSpace + ")??FOREIGN" + commentOrSpace + "DATA" + commentOrSpace + "WRAPPER" + commentOrSpace + id
```
[the remark this answers is not in the document]
Comment (rareddy) [reply pkg/util/vdbutil/parser.go:46]: added now

File: pkg/util/vdbutil/parser.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rareddy) [reply pkg/util/vdbutil/parser.go]: Added a test case showing the nested schema scenario

**Comment by rareddy** [comment]
@shawkins see the updated PR

**Pull request merged**